crates/services/producer/src/block_producer/tests.rs

```diff
@@ -865,67 +956,49 @@ impl TestContextBuilder {
         .unwrap()
         .compress(&Default::default());
 
-        let db = MockDb {
-            blocks: Arc::new(Mutex::new(
-                vec![(self.prev_height, previous_block)]
-                    .into_iter()
-                    .collect(),
-            )),
-            consensus_parameters_version: 0,
-            state_transition_bytecode_version: 0,
-        };
+        Arc::new(Mutex::new(HashMap::from_iter(Some((height, block)))))
+    }
+
+    fn build(self) -> TestContext<MockExecutor> {
+        let block_gas_limit = self.block_gas_limit.unwrap_or_default();
 
         let mock_relayer = MockRelayer {
             latest_block_height: self.latest_block_height,
             latest_da_blocks_with_costs: self.blocks_with_gas_costs.clone(),
             ..Default::default()
         };
 
+        let db = MockDb {
+            blocks: self.pre_existing_blockss(),
+            consensus_parameters_version: 0,
+            state_transition_bytecode_version: 0,
+        };
+
         TestContext {
             relayer: mock_relayer,
-            block_gas_limit: self.block_gas_limit.unwrap_or_default(),
+            block_gas_limit,
             ..TestContext::default_from_db(db)
         }
     }
 
-    fn build_with_executor<Ex>(&self, executor: Ex) -> TestContext<Ex> {
-        let da_height = self.prev_da_height;
-        let previous_block = PartialFuelBlock {
-            header: PartialBlockHeader {
-                application: ApplicationHeader {
-                    da_height,
-                    ..Default::default()
-                },
-                consensus: ConsensusHeader {
-                    height: self.prev_height,
-                    ..Default::default()
-                },
-            },
-            transactions: vec![],
-        }
-        .generate(&[], Default::default())
-        .unwrap()
-        .compress(&Default::default());
-
-        let db = MockDb {
-            blocks: Arc::new(Mutex::new(
-                vec![(self.prev_height, previous_block)]
-                    .into_iter()
-                    .collect(),
-            )),
-            consensus_parameters_version: 0,
-            state_transition_bytecode_version: 0,
-        };
+    fn build_with_executor<Ex>(self, executor: Ex) -> TestContext<Ex> {
+        let block_gas_limit = self.block_gas_limit.unwrap_or_default();
 
         let mock_relayer = MockRelayer {
             latest_block_height: self.latest_block_height,
             latest_da_blocks_with_costs: self.blocks_with_gas_costs.clone(),
             ..Default::default()
         };
 
+        let db = MockDb {
+            blocks: self.pre_existing_blockss(),
```

Comment:
```suggestion
            blocks: self.pre_existing_blocks(),
```

Reply:
Hahah, sounds like a snake talking :sweat_smile: I'll fix. Thanks for spotting!

Reply:
Fixed in https://github.com/FuelLabs/fuel-core/pull/2206/commits/c8f3ee1b540bf35a0cf2fc3cd8ed9fa48bb9223e